pkg/bindings/images/build.go

```diff
@@ -643,11 +636,154 @@ func Build(ctx context.Context, containerFiles []string, options types.BuildOpti
 		}
 	}()
 
+	var requestBody io.Reader
+	var contentType string
+
+	if len(options.AdditionalBuildContexts) > 0 {
+		logrus.Debug("Checking for current version (Has to have 5.6.0+)")
+
+		supported, err := serverSupportsMultipartBuildContexts(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("checking server capabilities: %w", err)
+		}
+
+		if !supported {
+			return nil, fmt.Errorf("server does not support additional build contexts (requires Podman 5.5.2 or newer)")
+		}
+
+		logrus.Debugf("Using additional build contexts: %v", options.AdditionalBuildContexts)
+
+		pipeReader, pipeWriter := io.Pipe()
+		writer := multipart.NewWriter(pipeWriter)
+		contentType = writer.FormDataContentType()
+		requestBody = pipeReader
+
+		go func() {
+			defer pipeWriter.Close()
+			defer writer.Close()
+
+			if tarfile != nil {
+				mainPart, err := writer.CreateFormFile("context", "context.tar")
+				if err != nil {
+					pipeWriter.CloseWithError(err)
+					return
+				}
+
+				if _, err := io.Copy(mainPart, tarfile); err != nil {
+					pipeWriter.CloseWithError(err)
+					return
+				}
+			}
+
+			for name, context := range options.AdditionalBuildContexts {
+				logrus.Debugf("Processing additional build context: %q", name)
+
+				switch {
+				case context.IsImage:
+					metadataField := fmt.Sprintf("buildcontext-image-%s", name)
+					if err := writer.WriteField(metadataField, context.Value); err != nil {
+						pipeWriter.CloseWithError(fmt.Errorf("adding URL context %q: %w", name, err))
+						return
+					}
+					logrus.Debugf("Added image context %q: %s", name, context.Value)
+
+				case context.IsURL:
+					metadataField := fmt.Sprintf("buildcontext-url-%s", name)
+					if err := writer.WriteField(metadataField, context.Value); err != nil {
+						pipeWriter.CloseWithError(fmt.Errorf("adding URL context %q: %w", name, err))
+						return
+					}
+					logrus.Debugf("Added URL context %q: %s", name, context.Value)
+
+				default:
+					part, err := writer.CreateFormFile(fmt.Sprintf("buildcontext-local-%s", name), name+".tar")
+					if err != nil {
+						pipeWriter.CloseWithError(err)
+						return
+					}
+
+					// local context is already a tar
+					if strings.HasSuffix(context.Value, ".tar") {
```

Comment:
Expecting the file name to correctly indicate things can lead to errors.  Use `github.com/containers/storage/pkg/archive.IsArchivePath()`.